pkg/controller/data_upload_controller.go

```diff
@@ -292,19 +292,22 @@ func (r *DataUploadReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	} else if du.Status.Phase == velerov2alpha1api.DataUploadPhaseInProgress {
 		log.Info("Data upload is in progress")
 		if du.Spec.Cancel {
+			log.Info("Data upload is being canceled")
+
 			fsBackup := r.dataPathMgr.GetAsyncBR(du.Name)
 			if fsBackup == nil {
+				r.OnDataUploadCancelled(ctx, du.GetNamespace(), du.GetName())
 				return ctrl.Result{}, nil
 			}
-			log.Info("Data upload is being canceled")
 
-			// Update status to Canceling.
+			// Update status to Canceled
 			original := du.DeepCopy()
-			du.Status.Phase = velerov2alpha1api.DataUploadPhaseCanceling
+			du.Status.Phase = velerov2alpha1api.DataUploadPhaseCanceled
```

Comment:
The same as DD, rollback to the original code and wait `fsBackup` to cancel itself

Reply:
Done